core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -3270,6 +3285,13 @@ object ZIO {
       zio.provideSome[R2](r2 => r2.add(r1))
```

Comment:
`provideOne` can be trivially implemented in terms of `provideSomeLayer` now. It isn't being used anywhere. We should either delete if we want layers to be the sole means of doing this or move up to the top level with new naming convention.

Reply:
Agreed, let's delete it.